Mage/src/main/java/mage/watchers/common/ManaPaidObjectSourceWatcher.java

```diff
@@ -0,0 +1,54 @@
+package mage.watchers.common;
+
+import mage.MageObject;
+import mage.MageObjectReference;
+import mage.constants.WatcherScope;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.events.ManaPaidEvent;
+import mage.watchers.Watcher;
+
+import java.util.*;
+
+/**
+ * @author Susucr
+ */
+public class ManaPaidObjectSourceWatcher extends Watcher {
```

Comment:
Why not improved `ManaPaidSourceWatcher`? It has both mana source and dest source objects to control:

![shot_231119_214228](https://github.com/magefree/mage/assets/8344157/7149b5da-7b8b-4556-9143-0d8d41e7d382)

Reply:
This is tracking a lot of Objects per mana payment, I didn't want to include the map in all games. Here it is restricted to games where cards with the triggers are used.